docs/bin/dump_config.py

```diff
@@ -62,11 +63,10 @@ def main(args):
     output_name = os.path.join(output_dir, template_file.replace('.j2', ''))
     temp_vars = {'config_options': config_options}
 
-    with open(output_name, 'wb') as f:
-        f.write(template.render(temp_vars).encode('utf-8'))
+    data = template.render(temp_vars).encode('utf-8')
 
-    return 0
```

Comment:
```return 0``` and ```sys.exit(main(sys.argv[:]))``` should remain.

Keeping those allows the script to return a different exit code on error and isn't unusual or confusing for someone who is used to UNIX programming.

Reply:
Well, none of the other scripts are doing this. And it adds absolutely no value as this script was not designed to return anything else whatsoever. So this is nothing more than a mental exercise.

So you want me to add this to the other scripts ? Do we also add stuff like this to modules ? I hope we don't.

Reply:
We could add it to other scripts.

modules have their own exit points (fail_json and exit_json) which already call sys.exit(1) and sys.exit(0) respectively.

Reply:
Note though, I wouldn't add to other scripts in this PR.... it's a separate code cleanup from the main purpose of this one (to stop rebuilding already-built docs)

Reply:
No, I think it's useless but leave it up to @webknjaz to rewrite those scripts as he clearly sees great benefits :-)